go-runtime/ftl/option.go

```diff
@@ -165,6 +165,25 @@ func (o Option[T]) Default(value T) T {
 	return value
 }
 
+func (o Option[T]) MarshalJSON() ([]byte, error) {
+	if o.ok {
+		return json.Marshal(o.value)
+	}
+	return []byte("null"), nil
+}
+
+func (o *Option[T]) UnmarshalJSON(data []byte) error {
+	if string(data) == "null" {
+		o.ok = false
+		return nil
+	}
+	if err := json.Unmarshal(data, &o.value); err != nil {
```

Comment:
- this used to be encoding.Unmarshal(...)
- now that needs a context to work, which we don't have with standard json unmarshalling
- Would this change mean that ftl.option can work with standard json marshalling, but without some of our enum/sumtype abilities?